.github/workflows/validate.yml

```diff
@@ -22,17 +22,24 @@ jobs:
               uses: actions/setup-python@v4
               with:
                   python-version: '3.10'
-                  cache: pip
-                  cache-dependency-path: pyproject.toml
+
+            - name: Set up uv
+              uses: astral-sh/setup-uv@v3
+              with:
+                  version: 0.4.x
+                  enable-cache: true
 
             - name: Install dependencies
-              run: pip install .
+              run: |
+                  uv sync --locked --all-extras --dev
```

Comment:
I guess I should read the manual on what `uv` actually does here, but if it works, it works.

Reply:
Just syncs the current environment with the dependencies in the lock file. For this repo I don't think we need to bother with too much dependency auto updates but we could add a scheduled job that updates the lock